torch/testing/_internal/common_quantization.py

```diff
@@ -101,6 +106,99 @@ def test_only_train_fn(model, train_data, loss_fn=_default_loss_fn):
             correct += (predicted == target).sum().item()
     return train_loss, correct, total
 
+class AverageMeter(object):
+    """Computes and stores the average and current value"""
+    def __init__(self, name, fmt=':f'):
+        self.name = name
+        self.fmt = fmt
+        self.reset()
+
+    def reset(self):
+        self.val = 0
+        self.avg = 0
+        self.sum = 0
+        self.count = 0
+
+    def update(self, val, n=1):
+        self.val = val
+        self.sum += val * n
+        self.count += n
+        self.avg = self.sum / self.count
+
+    def __str__(self):
+        fmtstr = '{name} {val' + self.fmt + '} ({avg' + self.fmt + '})'
+        return fmtstr.format(**self.__dict__)
+
+
+def accuracy(output, target, topk=(1,)):
+    """Computes the accuracy over the k top predictions for the specified values of k"""
+    with torch.no_grad():
+        maxk = max(topk)
+        batch_size = target.size(0)
+
+        _, pred = output.topk(maxk, 1, True, True)
+        pred = pred.t()
+        correct = pred.eq(target.view(1, -1).expand_as(pred))
+
+        res = []
+        for k in topk:
+            correct_k = correct[:k].view(-1).float().sum(0, keepdim=True)
+            res.append(correct_k.mul_(100.0 / batch_size))
+        return res
+
+def train_one_epoch(model, criterion, optimizer, data_loader, device, ntrain_batches):
+    model.train()
+    top1 = AverageMeter('Acc@1', ':6.2f')
+    top5 = AverageMeter('Acc@5', ':6.2f')
+    avgloss = AverageMeter('Loss', '1.5f')
+
+    cnt = 0
+    for image, target in data_loader:
+        start_time = time.time()
+        print('.', end='')
+        cnt += 1
+        image, target = image.to(device), target.to(device)
+        output = model(image)
+        loss = criterion(output, target)
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+        acc1, acc5 = accuracy(output, target, topk=(1, 5))
+        top1.update(acc1[0], image.size(0))
+        top5.update(acc5[0], image.size(0))
+        avgloss.update(loss, image.size(0))
+        if cnt >= ntrain_batches:
+            print('Loss', avgloss.avg)
+
+            print('Training: * Acc@1 {top1.avg:.3f} Acc@5 {top5.avg:.3f}'
+                  .format(top1=top1, top5=top5))
+            return
+
+    print('Full imagenet train set:  * Acc@1 {top1.global_avg:.3f} Acc@5 {top5.global_avg:.3f}'
+          .format(top1=top1, top5=top5))
+    return
+
+def ddp_setup(rank, world_size):
```

Comment:
there should be some utilities in `test/distributed` for this, can we reuse?

Reply:
you mean https://github.com/pytorch/pytorch/blob/master/test/distributed/test_distributed.py? it looks like they are doing everything inline, I didn't see a function. also I'm not sure how to reuse this https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_distributed.py#L207

Reply:
np, can we split this into a separate file then?  Ideally people who look at `common_quantization.py` shouldn't have to read code related to DDP.

Reply:
Sure

Reply:
you mean split this to common_quantization_ddp or something?

Reply:
I think common_quantization just contains all utility functions related to quantization tests, looking at other common_* code, i don't feel we should create a new file just for this

Reply:
I accepted the PR, I trust you.  I do feel like DDP testing ideally should be fully separate, and we shouldn't be testing every single thing with DDP but instead making sure that DDP team supports fx properly, and trusting their system to work.  It's a high level question of do we test A*B*C things (test everything, but more expensive to maintain), or do we test A*B things and B*C things with clean and well defined APIs and trust that A*B*C works by induction from A*B and B*C working.

Reply:
> I think common_quantization just contains all utility functions related to quantization tests, looking at other common_* code, i don't feel we should create a new file just for this

also an optional suggestion.  Usually after some LOC (~200-400LOC), it's harder for a code reader to fully understand the entire file quickly.  Keeping files small and single purposed allows for easier code browsing.

Reply:
I agree we don't need to test everything, but unit test and integration tests are both important, in this case we are testing resnet18 with ddp and it is a integration test that makes sure ddp + quantization + fx works, this can not be tested in unit test only I think.

re code readability, I agree we don't want to mix unrelated things in a single file. but common_quantization is more of a util file where we put all util functions in. I think we can split the models to a separate file in common_quantization.py. but the other code are utility functions used in test.